Debugging nodejs / nvm

coreyferguson opened the pull request:
I manually modified the SUMMARY.md as I noticed several differences from the one auto-generated:
- URL's began with `/`
- Spacing differences

I've made this change into a separate commit in case that's not right.

I've only added information from linux as that's the only environment I've tested on. Also I didn't see the `install-drivers` section on other OS's. I guess it's not required for them?

I assume these instructions would work with Mac OS. Not sure about Windows.

HipsterBrown commented on Debugging/NodeJs.md:
```diff
@@ -0,0 +1,29 @@
+
+Installation instructions of the Tessel CLI can be found here: http://tessel.github.io/t2-start/
+
+This document will address common problems related to NodeJs usage.
```
NodeJs should be changed to Node.js.

HipsterBrown commented on Debugging/NodeJs.md:
```diff
@@ -0,0 +1,29 @@
+
+# NodeJs
+
+Installation instructions of the Tessel CLI can be found here: http://tessel.github.io/t2-start/
```
Can we make the url into an official markdown link?

`[http://tessel.github.io/t2-start/](http://tessel.github.io/t2-start/)`

Also, I believe the text should be updated to read:

Installation instructions **for** the Tessel CLI can be found here:

I highlighted the changed word in bold.

HipsterBrown commented on Debugging/NodeJs.md:
```diff
@@ -0,0 +1,29 @@
+
+### Linux
+
+Install `t2-cli` module as normal. To install the drivers, perform the following:
```
Can we add a bit more information here about why the drivers need to be installed separately? Maybe @johnnyman727 can provide some context around that.

kevinmehall replied:
On Linux, they're not technically "drivers", but [udev rules](https://www.kernel.org/pub/linux/utils/kernel/hotplug/udev/udev.html) that are installed under `/etc/` to set the permissions so that your normal user account can access the device. This can only be installed as root. You normally don't (or shouldn't) `npm install` as root, and even if you do, it drops privileges to run the build script.

On Windows, the WinUSB driver should be installed automatically from Windows Update when you plug in the device for the first time. Node / the `t2` command are not involved and you don't need to `t2 install-drivers`.

coreyferguson replied:
> ... but udev rules that are installed under /etc/ to set the permissions so that your normal user account can access the device. This can only be installed as root.

I wonder if the script itself can request sudo access after it's run?

rwaldron commented:
> Also I didn't see the install-drivers section on other OS's. I guess it's not required for them?

That's correct

coreyferguson commented:
Updated PR...

> I'm not sure about manually modifying the SUMMARY.md because it will be overwritten the next time the docs are built, so it might be best to revert that commit.

Done. The updated (auto-generated) SUMMARY.md changes are included as part of [4f4edf5c4934da03edaf374ef31153ea13d5c200](https://github.com/tessel/t2-docs/pull/125/commits/4f4edf5c4934da03edaf374ef31153ea13d5c200).

> Can we make the url into an official markdown link? ... Installation instructions **for** the Tessel CLI can be found here ...

Done. Changes found here: [630926120d0d893c80b51891364320b65e485dd2](https://github.com/tessel/t2-docs/pull/125/commits/630926120d0d893c80b51891364320b65e485dd2).

> NodeJs should be changed to Node.js.

Done. Text changes here: [630926120d0d893c80b51891364320b65e485dd2](https://github.com/tessel/t2-docs/pull/125/commits/630926120d0d893c80b51891364320b65e485dd2). And I renamed the file itself so the SUMMARY.md reflects this verbiage as well here: [03dd2104cc54488152c2bf4ec1bf5c016190643f](https://github.com/tessel/t2-docs/pull/125/commits/03dd2104cc54488152c2bf4ec1bf5c016190643f)

Frijol commented:
This looks good, thanks!
